java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java

```diff
@@ -1172,37 +1207,63 @@ public Boolean visitBreak(BreakTree node, ConstructorData p) {
 
             Tree target = info.getTreeUtilities().getBreakContinueTargetTree(getCurrentPath());
             
-            resumeAfter(target, variable2State);
+            breakTo(target);
 
-            variable2State = new HashMap< Element, State>();
+            return null;
+        }
+
+        public Boolean visitYield(Tree node, ConstructorData p) {
+            scan(TreeShims.getYieldValue(node), p);
+
+            Tree target = info.getTreeUtilities().getBreakContinueTargetTree(getCurrentPath());
             
+            breakTo(target);
+
             return null;
         }
 
+        private void breakTo(Tree target) {
+            resumeAfter(target, variable2State);
+
+            variable2State = new HashMap< Element, State>();
```

Comment:
Jackpot:
warning: Redundant type arguments in new expression (use diamond operator instead).